metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -309,14 +313,18 @@ void CanonicalDBG::call_incoming_kmers(node_index node,
         assert(c == get_node_sequence(prev)[0]);
         assert(c == boss::BOSS::kSentinel || traverse_back(node, c) == prev);
 
-        if (c != boss::BOSS::kSentinel)
+        if (c != boss::BOSS::kSentinel) {
             callback(prev, c);
-
-        parents[s] = prev;
+            parents[s] = prev;
+            --max_num_edges_left;
+        }
     });
 
-    if (parents[alphabet_encoder_[boss::BOSS::kSentinel]])
-        callback(parents[alphabet_encoder_[boss::BOSS::kSentinel]], boss::BOSS::kSentinel);
+    if (has_sentinel_ && parents[alphabet_encoder_[boss::BOSS::kSentinel]]
+            && max_num_edges_left + has_sentinel_ == alphabet.size()) {
+        callback(parents[alphabet_encoder_[boss::BOSS::kSentinel]],
+                 boss::BOSS::kSentinel);
+    }
```

Comment:
ah, yeah. Makes sense. Because it might have a dummy sink on the forward strand but the reverse strand doesn't have any suffix, so no dummy source would be called.